backend/test/lastwills.e2e-spec.ts

```diff
@@ -0,0 +1,435 @@
+import { getConnectionToken } from '@m8a/nestjs-typegoose'
+import { HttpStatus, INestApplication, ValidationPipe } from '@nestjs/common'
+import { ConfigModule, ConfigService } from '@nestjs/config'
+import { JwtService } from '@nestjs/jwt'
+import { Test, TestingModule } from '@nestjs/testing'
+import { Connection, Model } from 'mongoose'
+import * as request from 'supertest'
+import { RefreshJWTPayload } from '../src/auth/interfaces/refresh-jwt-payload.interface'
+import { DbModule } from '../src/db/db.module'
+import { LastWill, LastWillMetadata } from '../src/db/entities/lastwill.entity'
+import { User } from '../src/db/entities/users.entity'
+import { LastWillsModule } from '../src/last-wills/lastwills.module'
+import { paymentPlans } from '../src/payments/interfaces/payments'
+import { SharedModule } from '../src/shared/shared.module'
+import { MockConfigService } from './helpers/config-service.helper'
+import {
+  closeInMongodConnection,
+  rootTypegooseTestModule,
+} from './helpers/mongo.helper'
+import {
+  SAMPLE_USER,
+  SAMPLE_USER_PW_HASH,
+  sampleObject,
+} from './helpers/sample-data.helper'
+
+describe('LastWillsController (e2e)', () => {
+  let app: INestApplication
+  let jwtService: JwtService
+  let connection: Connection
+  let userModel: Model<User>
+  let lastWillsModel: Model<LastWill>
+  let configService: ConfigService
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        DbModule,
+        SharedModule,
+        ConfigModule.forRoot({ isGlobal: true }),
+        rootTypegooseTestModule(),
+        LastWillsModule,
+      ],
+    })
+      .overrideProvider(ConfigService)
+      .useClass(MockConfigService)
+      .compile()
+
+    app = await moduleFixture.createNestApplication()
+    app.useGlobalPipes(
+      new ValidationPipe({
+        whitelist: true,
+        transform: true,
+        transformOptions: { enableImplicitConversion: true },
+      }),
+    )
+
+    jwtService = app.get<JwtService>(JwtService)
+    connection = await app.get(getConnectionToken())
+    configService = app.get<ConfigService>(ConfigService)
+    userModel = connection.model<User>('User')
+    lastWillsModel = connection.model<LastWill>('LastWill')
+
+    await app.init()
+  })
+
+  afterAll(async () => {
+    await closeInMongodConnection()
+    await app.close()
+  })
+
+  let token
+  let user: User
+  beforeEach(async () => {
+    await userModel.deleteMany({})
+    await lastWillsModel.deleteMany({})
+    user = await userModel.create({
+      ...SAMPLE_USER,
+      password: await SAMPLE_USER_PW_HASH(),
+    })
+    token = jwtService.sign(
+      {
+        id: user._id,
+        email: user.email,
+      } as RefreshJWTPayload,
+      { secret: configService.get('JWT_SECRET') },
```

Comment:
So this is a refreshjwt but you are not using the refresh jwt secret to sign it? 🤔